flake.nix

```diff
@@ -70,6 +84,66 @@
               }
         */
         readOnlyPkgs = ./nixos/modules/misc/nixpkgs/read-only.nix;
+
+        /*
+          Replaces the module for `nixpkgs.*` by a simpler module that can
+          reuse `legacyPackages`, but does not have any backwards compatibility,
+          so no `nixpkgs.*System`, etc.
+
+          Ignored when `readOnlyPkgs` is imported.
+         */
+        noLegacyPkgs = {
+          imports = [ ./nixos/modules/misc/nixpkgs/no-legacy.nix ];
+          nixpkgs._memoize = pkgsMemoizer;
+        };
+      };
+
+      modules = {
+        /* Modules that are not specific to any application of the module system. */
+        generic = {
+          nixpkgs = {
+            imports = [
+              ./pkgs/top-level/module/module.nix
+            ];
+            config._memoize = pkgsMemoizer;
+          };
+        };
+      };
+
+      /*
+        https://flake.parts
+        Maintainer: @roberth
+      */
+      flakeModules = {
+        pkgs = { flake-parts-lib, ... }: {
```

Comment:
> * This can be turned into N+M by creating stable standards and using them instead.

flake-parts is such an interface.

> * flake-parts sould either be moved to the NixOS GitHub organization, or

I don't think that would help the project. Maintaining it is 95% maintaining a clean interface (as you alluded to), pushing back on things that don't belong in it. I wouldn't trust someone else to do that.
If for whatever reason the NixOS organization needs to take ownership of this project, it can be forked. Until then I would recommend them to delegate this work to me.

> * code can't easily be tested and documented without also involving flake-parts

I run CI for it asynchronously, twice a day and I'm happy to fix things as they come up. There's very little to integrate, so problems will be rare, based on my experience with other integrations. It's a very thin layer on top of the generic module after all.

Would you like to review the rest of the PR? I think it's been decently covered by John and Valentin, but the part about optional NixOS modules in the manual is still new, as is the `noLegacyPkgs` module.

Reply:
I've removed the flake-parts module for now. Flake-parts users will still be able to use the generic pkgs module.